File: C-Sharp/Desafio04-C-Sharp/Desafio04-C-Sharp/com.gft.model/VideoGame.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Desafio04_C_Sharp.com.gft.model
{
    class VideoGame:Produto
    {
        public string Marca { get; set; }
        public string Modelo { get; set; }
        public bool IsUsado { get; set; }

        public VideoGame() { }

        public VideoGame(string nome, double preco, int qtd, string marca, string modelo, bool isUsado):base(nome, preco, qtd)
        {
            Marca = marca;
            Modelo = modelo;
            IsUsado = isUsado;
        }

        public double calculaImposto()
        {
            double imposto;
            if (IsUsado == true)
            {
                imposto = Preco * 0.25;
                Console.WriteLine("Imposto {0} {1} usado: R$ {2:0.00}", Nome, Modelo, imposto);
            }
            else
            {
                imposto = Preco * 0.45;
                Console.WriteLine("Imposto {0} {1}: R$ {2:0.00}", Nome, Modelo, imposto);
            }
            return imposto;
        }
    }
}

File: Java/Desafio4 - GFT/src/br/com/gft/model/VideoGame.java
package br.com.gft.model;

import br.com.gft.interfaces.Imposto;

public class VideoGame extends Produto implements Imposto {

	private String marca, modelo;
	private boolean isUsado;
	
	public VideoGame() {
		super();
	}
	
	public VideoGame(String nome, double preco, int qtd, String marca, String modelo, boolean isUsado) {
		super(nome, preco, qtd);
		setMarca(marca);
		setModelo(modelo);
		setUsado(isUsado);
	}
	
	public String getMarca() {
		return marca;
	}
	public void setMarca(String marca) {
		this.marca = marca;
	}
	public String getModelo() {
		return modelo;
	}
	public void setModelo(String modelo) {
		this.modelo = modelo;
	}
	public boolean getUsado() {
		return isUsado;
	}
	public void setUsado(boolean isUsado) {
		this.isUsado = isUsado;
	}

	
	@Override
	public double calculaImposto() {
		double imposto;
		if (getUsado() == true) {
			imposto = getPreco() * 0.25;
			System.out.printf("Imposto %s %s usado: R$ %.2f \n", getNome(), getModelo(), imposto);
		} else {
			imposto = getPreco() * 0.45;
			System.out.printf("Imposto %s %s: R$ %.2f", getNome(), getModelo(), imposto);
		}
		return imposto;
	}
	
	
	
}
